### PR TITLE
add JSONWebKeySet in attestation.json file

### PR DESCRIPTION
**Pull request by shleiAmy**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json [hunk 1]
```diff
@@ -423,6 +423,10 @@
         "attestationPolicy": {
           "type": "string",
           "description": "Name of attestation policy."
+                },
```
Comment (MyronFanQiu) [inline specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json]: would you mind correcting the format?

File: specification/common-types/resource-management/v1/types.json [hunk 2]
```diff
@@ -319,6 +319,110 @@
...
+        "kid",
+        "alg"
+      ],
+      "properties": {
+        "alg": {
+          "description": "The \"alg\" (algorithm) parameter identifies the algorithm intended for\nuse with the key.  The values used should either be registered in the\nIANA \"JSON Web Signature and Encryption Algorithms\" registry\nestablished by [JWA] or be a value that contains a Collision-\nResistant Name.",
+          "type": "string",
+          "x-go-name": "Alg"
+        },
+        "crv": {
```
Comment (MyronFanQiu) [inline specification/common-types/resource-management/v1/types.json]: It's quite confusing to have such naming pattern. Is it possible to use longer words? at least, we should add description.

**Review by mmyyrroonn — CHANGES_REQUESTED** [review]

File: specification/common-types/resource-management/v1/types.json [hunk 3]
```diff
@@ -319,6 +319,110 @@
         }
       },
       "description": "The resource management error additional info."
+    },
+    "JSONWebKey": {
```
Comment (KrisBash) [inline specification/common-types/resource-management/v1/types.json]: Why is this going in common-types? It does not appear to be universal to all RPs?

**Review by KrisBash — CHANGES_REQUESTED** [review]
This looks like a misunderstanding about common-types. Please let me know if there's a reason this definition should be shared across RPs

File: specification/common-types/resource-management/v1/types.json (hunk 3 above)
Comment (shleiAmy) [reply specification/common-types/resource-management/v1/types.json]: I will make the JSONWebKey definition in our own RPs.

File: specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json (hunk 1 above)
Comment (shleiAmy) [reply specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json]: May I ask what's the format you were talking about? @MyronFanQiu

File: specification/common-types/resource-management/v1/types.json (hunk 3 above)
Comment (LarryOsterman) [reply specification/common-types/resource-management/v1/types.json]: Where do protocol elements defined in an IETF RFC go? I would think that elements defined in an RFC are by definition "common" and are not protocol specific. If there's a better place under common-types than types.json for this definition, please let us know.

**Comment by LarryOsterman** [comment]
> This looks like a misunderstanding about common-types. Please let me know if there's a reason this definition should be shared across RPs

Asked earlier, but: Where do types defined by an IETF RFC go? These types by definition are common to all protocols implementing that RFC. It seems like "common-types" is a reasonable location for such types. Guidance for how we handle RFC defined types would be appreciated.

File: specification/common-types/resource-management/v1/types.json (hunk 3 above)
Comment (LarryOsterman) [reply specification/common-types/resource-management/v1/types.json]: Oh, and Amy: Please don't make the changes to turn these definitions private until we get clear guidance that IETF defined types are not "common" types (or guidance about where we should put internet standard type definitions). It seems rather strange that the SDK requires private definitions of these standardized types.

Comment (KrisBash) [reply specification/common-types/resource-management/v1/types.json]: @LarryOsterman  "common" in this case is a reference to cross-RP usage patterns. Are there other RPs that will be using this type in the foreseeable future?

Comment (LarryOsterman) [reply specification/common-types/resource-management/v1/types.json]: I don't know the answer to your question. In general, standards are intended to be used by multiple customers, so everyone who implements RFC7517 or RFC7515 (which depends on 7515) is likely to require this. These are widely used internet protocols, so it seems likely that there are multiple teams that would require this. I do know that AAD and AKV for certain use this protocol and I'm willing to bet there are many more azure services that are going to use them. However i don't know if those services are resource providers.

If I was designing the layout for files in the azure SDK, I would have an internet standards folder indexed by RFC number which would host the swagger documents for internet protocols. That way any RPs which want to implement JSON based internet protocols would have a place to put their swagger documents.

Comment (LarryOsterman) [reply specification/common-types/resource-management/v1/types.json]: PS: I also suspect that it is highly likely that one or more azure services implement internet standard protocols So having support in the azure SDK for internet protocols seems like a good thing.

**Comment by MyronFanQiu** [comment]
@KrisBash Hello. Could you give the guidance to @shleiAmy? @shleiAmy For the failures in the spellcheck, you can update this file by adding the words into it. https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt

**Comment by shleiAmy** [comment]
@MyronFanQiu I have resubmitted pr to fix the word check error. Thanks

File: specification/common-types/resource-management/v1/types.json (hunk 3 above)
Comment (anthony-c-martin) [reply specification/common-types/resource-management/v1/types.json]: I'm in favor of making this reusable, but agree with @KrisBash's point about the specific file it's going into - `.../resource-management/v1/types.json` are  generally types owned by ARM relating to resources.

What about creating a new file for it in a different location - `specification/common-types/rfcs/rfc7517.json` (or something more user friendly)?

@MyronFanQiu what are your thoughts?

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/common-types/resource-management/v1/types.json (hunk 3 above)
Comment (KrisBash) [reply specification/common-types/resource-management/v1/types.json]: @LarryOsterman @MyronFanQiu  I am ok with the proposal Larry and Anthony suggested of having a common definition directory for rfcs. I think it's not strictly necessary at this point as I am not aware of other RPs using this type, but it's not a bad precedent. I am opposed to modifying types.json. Ownership of this file needs to be retained by ARM as changes to this file can be very disruptive to many providers.

Comment (LarryOsterman) [reply specification/common-types/resource-management/v1/types.json]: I'm totally happy with anthony's suggestion.

And there are already RPs using JSON Web Keys in their protocols.

For instance I found
specification\containerregistry\data-plane\Microsoft.ContainerRegistry\preview\2018-08-10\containerregistry.json 
which has a private definition for JWKs. And AKV uses x5c in its protocol elements.

My suspicion is that they ran into the same issue as flagged here and they weren't as obnoxious about API private definitions.

File: specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json [hunk 4]
```diff
@@ -423,8 +423,128 @@
...
+        "use",
+        "kty",
+        "kid",
+        "alg"
+      ],
+      "properties": {
+        "alg": {
+          "description": "The \"alg\" (algorithm) parameter identifies the algorithm intended for\nuse with the key.  The values used should either be registered in the\nIANA \"JSON Web Signature and Encryption Algorithms\" registry\nestablished by [JWA] or be a value that contains a Collision-\nResistant Name.",
+          "type": "string",
+          "x-go-name": "Alg"
```
[the remark this answers is not in the document]
Comment (shleiAmy) [reply specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json]: x-go-name & x-go-package are removed

File: specification/common-types/resource-management/v1/types.json (hunk 2 above)
Comment (shleiAmy) [reply specification/common-types/resource-management/v1/types.json]: description are added

File: specification/common-types/resource-management/v1/types.json (hunk 3 above)
Comment (shleiAmy) [reply specification/common-types/resource-management/v1/types.json]: JSONWebKeySet definition has been moved to a specific folder: specification/common-types/rfcs/rfc7517.json

File: specification/common-types/rfcs/rfc7517.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    "JSONWebKeySet": {
+      "type": "object",
+      "properties": {
+        "keys": {
+          "description": "The value of the \"keys\" parameter is an array of JWK values.  By\ndefault, the order of the JWK values within the array does not imply\nan order of preference among them, although applications of JWK Sets\ncan choose to assign a meaning to the order for their purposes, if\ndesired.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JSONWebKey"
+          },
+          "x-go-name": "Keys"
```
Comment (anthony-c-martin) [inline specification/common-types/rfcs/rfc7517.json]: [nit] remove the x-go-name property

**Review by anthony-c-martin — APPROVED** [review]

File: specification/common-types/rfcs/rfc7517.json (hunk 5 above)
Comment (shleiAmy) [reply specification/common-types/rfcs/rfc7517.json]: removed

**Review by mmyyrroonn — APPROVED** [review]

**Comment by MyronFanQiu** [comment]
@KrisBash Hello. Are you ok with this PR? If so, please sign off it.

**Review by KrisBash — APPROVED** [review]
LGTM thanks

**Pull request merged**